lean/main/04_metam.lean

```diff
@@ -639,6 +639,7 @@ Now, here are some examples of expressions in WHNF.
 Constructor applications are in WHNF (with some exceptions for numeric types):
 -/
 
+set_option pp.explicit false
```

Comment:
I think the real issue here is [above on line 569](https://github.com/leanprover-community/lean4-metaprogramming-book/pull/146/files#diff-4c9daca2cffec9eda3aa306e90e6f71a7c0e7b51e7c5d8387c71af9fd1cfa3a7R569), and specifically that that line should have been doing

`set_option pp.explicit true in`

(with the `in`) rather than globally changing the setting?

Could you perhaps try that instead and check whether that makes everything look correct?

Meanwhile, as someone who doesn't know much at all about metaprogramming, I find this example confusing to start off with -- the text just got finished explaining what the general form of something in WHNF is, and then here immediately starts with an example that prints differently? Naively to me it definitely seems worthwhile to explain that the pretty printer can make something look not WHNF even when it is, but *starting* with such an example seems weird if I follow what the text is saying.

But it does seem like the intention is to show `[1]` and that that's what it used to do before the change to 569, so I'm happy to merge if you take a look at the `in` bit above?

And thanks as usual!

Reply:
@Julian The change you suggested worked, thanks! PR updated.